nova_vm/src/ecmascript/abstract_operations/type_conversion.rs

```diff
@@ -790,28 +790,8 @@ pub(crate) fn to_big_int(
                 Ok(BigInt::from(0))
             }
         }
-        Primitive::String(idx) => {
-            let result = string_to_big_int(agent, idx.into());
-            let Some(result) = result else {
```

Comment:
**praise:** Good idea to move the throws into the `string_to_big_int` function! <3